docs/pages/Cardinal/API-Reference/Transactions.mdx

```diff
@@ -0,0 +1,208 @@
+# Transactions
+
+This API reference will help you understand how to use transactions and configuration options available for transactions.
+
+```go
+package transactions
+
+import "pkg.world.dev/world-engine/cardinal/ecs"
+
+// AttackInput is the input for the Attack transaction.
+type AttackInput struct {
```

Comment:
Can we call this `AttackMsg`

Reply:
that deviates from the named type parameters of the `NewTransactionType` function signature:

```go
func NewTransactionType[In, Out any]
```
if we're okay with the mismatch i can make the change

Reply:
i think we should create a ticket to change that too

Reply:
created https://linear.app/arguslabs/issue/CAR-165/change-type-parameters-names-of-transactiontypes

Reply:
I commented on the ticket, but I recommend Msg and Result. I'd like to have Receipt be the combination of the Result and any errors.